test/fluree/db/query/misc_queries_test.clj

```diff
@@ -237,5 +237,13 @@
                   ["fluree:commit:sha256:bbstbkfuob2d73ubsc3xvqckbmn7z35esq5k7i3rolj7hed7li2ox"
                    :f/context
                    "fluree:memory://b6dcf8968183239ecc7a664025f247de5b7859ac18cdeaace89aafc421eeddee"]]
-                @(fluree/query db* {:select ['?s '?p '?o]
-                                    :where  [['?s '?p '?o]]}))))))))
+                 @(fluree/query db* {:select ['?s '?p '?o]
+                                     :where  [['?s '?p '?o]]})))))
+      (testing "Illegal reference queries"
+        (let [test-subject @(fluree/query db {:select ['?s '?p]
```

Comment:
Is it possible to have a query of the form `{:select ['?s] :where [['?s '?p <some-actual-ref-iri>]]}`? Either way I think it would be nice to have a test for that to make the intended behavior clear.

Reply:
No, that will (and should) throw an exception. If you want to specify that an object is an actual reference, you have to use the `@id` predicate:

```clojure
'{:select [?s]
  :where [[?s :id <some ref>]]}
```

I have added a test for that.